src/animated_tile.cpp

```diff
@@ -51,28 +42,16 @@ void DeleteAnimatedTile(TileIndex tile)
 void AddAnimatedTile(TileIndex tile)
 {
 	MarkTileDirtyByTile(tile);
-
-	for (const TileIndex *ti = _animated_tile_list; ti < _animated_tile_list + _animated_tile_count; ti++) {
-		if (tile == *ti) return;
-	}
-
-	/* Table not large enough, so make it larger */
-	if (_animated_tile_count == _animated_tile_allocated) {
-		_animated_tile_allocated *= 2;
-		_animated_tile_list = ReallocT<TileIndex>(_animated_tile_list, _animated_tile_allocated);
-	}
-
-	_animated_tile_list[_animated_tile_count] = tile;
-	_animated_tile_count++;
+	_animated_tiles.Include(tile);
 }
 
 /**
  * Animate all tiles in the animated tile list, i.e.\ call AnimateTile on them.
  */
 void AnimateAnimatedTiles()
 {
-	const TileIndex *ti = _animated_tile_list;
-	while (ti < _animated_tile_list + _animated_tile_count) {
+	const TileIndex *ti = _animated_tiles.Begin();
+	while (ti < _animated_tiles.End()) {
```

Comment:
Shouldn't this be `!=`? `<` for pointer arithmetic feels a bit weird?

Reply:
The while loop can shorten the tile list, so the `<` was probably some safeguard. I decided to leave that alone.